### PR TITLE
resolve Cluster Healthy Condition Message isn't empty

### PR DESCRIPTION
**Pull request by qiuming520**
resolve Cluster Healthy Condition Message isn't empty case

Signed-off-by: qiuming520 <qiuming_yewu@cmss.chinamobile.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/synchromanager/clustersynchro/cluster_monitor.go [hunk 1]
```diff
@@ -61,9 +61,10 @@ func (synchro *ClusterSynchro) checkClusterHealthy() {
 	}
 
 	synchro.startResourceSynchro()
-	if lastReadyCondition.Status == metav1.ConditionTrue {
-		// TODO: if lastReadyCondition.Message != "", need process
+	if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == "" {
```
Comment (Iceber) [inline pkg/synchromanager/clustersynchro/cluster_monitor.go:64]: This is an outdated TODO comment.

The message is always not empty, and the content of the message currently has two.
* "cluster health responded with ok"
* fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)

We can decide whether to update the healthy condition by determining the content of the message

eg.
```go
        message := "cluster health responded with ok"
        if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == message {
                return
        }

        if _, err := synchro.dynamicDiscoveryManager.GetAndFetchServerVersion(); err != nil {
                message = fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)
        }

        if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == message {
                // the ready status and message has not been modified,
                // to reduce the cluster status updates, do not update the `synchro.healthyCondition`.
                return
        }

        synchro.healthyCondition.Store(metav1.Condition{
                Type:               clusterv1alpha2.ClusterHealthyCondition,
                Status:             metav1.ConditionTrue,
                Reason:             clusterv1alpha2.ClusterHealthyReason,
                Message:            message,
                LastTransitionTime: metav1.Now().Rfc3339Copy(),
        })
```

Comment (qiuming520) [reply pkg/synchromanager/clustersynchro/cluster_monitor.go]: yes~this is good suggestion, I fixed it

File: pkg/synchromanager/clustersynchro/cluster_monitor.go [hunk 2]
```diff
@@ -61,11 +61,29 @@ func (synchro *ClusterSynchro) checkClusterHealthy() {
...
+
+	synchro.healthyCondition.Store(metav1.Condition{
+		Type:               clusterv1alpha2.ClusterHealthyCondition,
+		Status:             metav1.ConditionTrue,
+		Reason:             clusterv1alpha2.ClusterHealthyReason,
+		Message:            message,
+		LastTransitionTime: metav1.Now().Rfc3339Copy(),
+	})
+
 	condition := metav1.Condition{
```
Comment (Iceber) [inline pkg/synchromanager/clustersynchro/cluster_monitor.go:87]: The following duplicate logic can be removed

Comment (qiuming520) [reply pkg/synchromanager/clustersynchro/cluster_monitor.go]: fixed it ! Thank you

File: pkg/synchromanager/clustersynchro/cluster_monitor.go [hunk 3]
```diff
@@ -61,26 +61,29 @@ func (synchro *ClusterSynchro) checkClusterHealthy() {
...
-	}()
+
+	if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == message {
+		return
+	}
 
 	if _, err := synchro.dynamicDiscoveryManager.GetAndFetchServerVersion(); err != nil {
-		condition.Message = fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)
-		klog.ErrorS(err, "Failed to get cluster version", "cluster", synchro.name)
+		message = fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)
```
Comment (Iceber) [inline pkg/synchromanager/clustersynchro/cluster_monitor.go:78]: message and condition.Message are not the same string.

You can create the metav1.Condition before setting it to `synchro.healthyCondition`

Comment (qiuming520) [reply pkg/synchromanager/clustersynchro/cluster_monitor.go:70]: yes~ fixed it ! Thank you

**Pull request merged**